### DISCUSSION TITLE
[Task]: Is there any plan to add Unsupervised Continual Anomaly Detection with Contrastively-learned Prompt?

### DISCUSSION THREAD

**Discussion by fat-921** [open-edge-platform/anomalib#1940] [Feature Requests] [answered]
### What is the motivation for this task?

1.The current anomaly detection can only train one model for one class. What if only one model is trained for multiple classes?
2.How to add continuous learning to anomaly detection model training?

### Describe the solution you'd like

I saw a paper like Unsupervised Continual Anomaly Detection with Contrastively-learned Prompt in AAAI-2024 [[paper link]](https://ojs.aaai.org/index.php/AAAI/article/view/28153) [[code link]](https://github.com/shirowalker/UCAD/tree/main) , it's a great idea to work with, and it might be even better to add it to anomalib.

**Comment by samet-akcay** [comment]
This looks like an interesting approach. I'll add it to the feature request section with `Good First Issue` label in case someone  wants to work on it.

Thanks for sharing it!

**Reply by inbasperu** [reply]
Hey @samet-akcay, I'd be really excited to work on this feature.

**Reply by inbasperu** [reply, marked as answer]
Could you please set up the issue and assign it to me? 
Thank you!

**Reply by samet-akcay** [reply]
@inbasperu, thanks a lot for your interest! Sure, it's yours!
